dask/dataframe/io/parquet/core.py

```diff
@@ -473,6 +481,21 @@ def to_parquet(
     # Trim any protocol information from the path before forwarding
     path = fs._strip_protocol(path)
 
+    if overwrite:
+        working_dir = fs.expand_path(".")[0]
```

Comment:
I'm not sure why this needs to be checked. Do we end up in a back situation for local files if the current directory goes away (and then is immediately recreated) ? This would *only* be relevant to the local filesystem, and otherwise requires extra calls.

Reply:
It does seem like it'd be bad if the current directory is deleted along with all its contents.

Reply:
```
import os
os.mkdir("oi")
os.chdir("oi")
cwd = os.getcwd()
import shutil
shutil.rmtree(cwd)
os.mkdir(cwd)
print(os.listdir())
```
prints `[]`, no error. If you run it in ipython, it does crash horribly (because ipython is trying to save history somewhere, I think).

Reply:
I think I'm happier having the `overwrite` be more conservative.

Reply:
I am happier not making unnecessary fs calls for the special case :|

Reply:
>I am happier not making unnecessary fs calls for the special case :|

Agree that its always best to avoid fs calls when possible,  but the possibility of a user writing to `"./"` with `overwrite=True` does make me nervous :)  Perhaps the fact that the default is `overwrite=False` is enough to rationalize the perspective that a mistake like that would be the users problem?

Reply:
But this runs on the client right? So if you run this from a notebook and use "." as your path it would be bad. But I guess that'd be bad anyways.

Reply:
If you say overwrite the cwd, I'd say that's your own problem. Also, this doesn't prevent people from addressing the same path with a different string (usually possible), but it *does* prevent people that really want to write into the current directory, or perhaps a directory named ".", which is allowed by some file systems.

Reply:
Ok I am convinced! I guess the conscensus is to get rid of the check and error and add a warning to the docstring.